.github/workflows/ci-javascript.yaml

```diff
@@ -44,36 +44,41 @@ jobs:
       - name: Build
         run: yarn build
 
-  Check-Changed-Folders:
+  Should-Generate-Test-Wrappers:
     runs-on: ubuntu-latest
     outputs:
-      should-generate: ${{ env.should-generate }}
+      should_generate: ${{ env.should_generate }}
+      wrappers_branch: ${{ env.wrappers_branch }}
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-      - name: Check changed files
-        id: changed-files
-        uses: tj-actions/changed-files@v35
+      - name: Extract Origin Dev
+        id: extract
+        uses: actions/github-script@v5
         with:
-          files: |
-            packages/cli/*
-            packages/schema/*
-            packages/wasm/*
-      - id: set-output
-        run: echo "should-generate=${{steps.changed-files.outputs.any_changed}}" >> $GITHUB_ENV
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const prDescription = context.payload.pull_request.body;
+            const regex = /\/generate-test-cases\s+(\S+)/;
+            const match = regex.exec(prDescription);
+            if (match) {
+              console.log(`Extracted text: ${match[1]}`);
+              core.exportVariable('wrappers_branch', match[1]);
+              core.exportVariable('should_generate', 'true');
+            } else {
+              core.exportVariable('should_generate', 'false');
+            }
 
   Generate-Test-Wrappers:
-    needs: Check-Changed-Folders
-    if: ${{ needs.Check-Changed-Folders.outputs.should-generate == 'true' }}
+    needs: Should-Generate-Test-Wrappers
+    if: ${{ needs.Should-Generate-Test-Wrappers.outputs.should_generate == 'true' }}
     uses: ./.github/workflows/generate-modified-test-wrappers.yaml
     with:
       cli-path: /toolchain/packages/cli
       wasm-path: /toolchain/packages/wasm
-
+      harness-ref: ${{ needs.Should-Generate-Test-Wrappers.outputs.wrappers_branch }}
 
   Test-Core:
     runs-on: ubuntu-latest
-    needs: Generate-Test-Wrappers
```

Comment:
IDK this seems off... we only test the core wrappers if we "tell" the CI to do so via the special `/` command in the description? Seems like we'd want to do it if the core files change.

Reply:
> we only test the core wrappers if we "tell" the CI to do so via the special / command in the description?

yes

what I am trying to achieve here is to remove the need to create a new branch in the test harness when one of these files has changed.

it's worth mentioning that the tests will still run with the released wrappers (the ones that are generated via the `fetchWrappers` function).

based on your feedback, I would like to propose some approaches:

1.- (with the current changes) if these files are changed, we can make the bot comment something like: "Looks like you changed some files in the packages CLI, Schema, or WASM; remember to add the /generate-test-cases branch_name in your PR description.".

2.- another approach we might take is to keep the flow as it currently is, but the workflow will check if the branch exists in the test harness (remember, this branch will have the same name as the one in the toolchain); if it doesn't, then we can just build against master branch (this will require a custom script)

Reply:
after thinking a bit more, a third one (and maybe the easier one) might be a mix of both worlds: we generate the wrappers every time the files change (as it currently is), BUT we can define the branch that's going to be used in the test harness using the /generate-test-case command? meaning that you don't **need** to create the branch in the test harness if it is not necessary :thinking:

Reply:
Okay I'm better understanding the nuances here, hmmmm 🤔

Reply:
Just got off a call w/ Cesar, we feel we have a good plan laid out, will detail it here along with the changes incomming.